modules/stochastic_tools/src/utils/GaussianProcess.C

```diff
@@ -263,8 +279,10 @@ GaussianProcess::getGradient(RealEigenMatrix & inputs) const
     {
       const auto global_index = first_index + ii;
       _covariance_function->computedKdhyper(dKdhp, inputs, hyper_param_name, ii);
-      RealEigenMatrix tmp = alpha * dKdhp - _K_cho_decomp.solve(dKdhp);
-      grad_vec[global_index] = -tmp.trace() / 2.0;
+      // This is just temporary. Convert dKhp_matrix to tensor
+      auto tmp = torch::trace(torch::mm(alpha, dKdhp) - torch::cholesky_solve(dKdhp, _K_cho_decomp))
```

Comment:
This can be expensive if the database is large. We are doing another solve in the background I think.